### PR TITLE
[Android] Output error message to android log instead of stderr

### PR DESCRIPTION
**Pull request by gwang-msft**
**Description**: [Android] Output error message to android log instead of stderr

**Motivation and Context**
- When use minimal build with exception disabled on Android, the error messages of ORT_THROW/ORT_NOT_IMPLEMENTED/ORT_ENFORCE/ORT_THROW_EX are output to stderr, if the APP using ORT triggers these, the App will crash and usually we cannot see the error message, change the output of the error message to Android System log which will be captured by logcat
- For now seems iOS will redirect stderr/stdout to XCode console, so we probably don't need to change the output of error message on iOS to NSLog for now, added TODO in the code

### CODE REVIEW

File: include/onnxruntime/core/common/common.h [hunk 1]
```diff
@@ -77,6 +81,20 @@ using common::Status;
...
+#ifdef ORT_NO_EXCEPTIONS
+#if defined(__ANDROID__)
+#define ORT_PRINT_ERROR_MSG(msg) \
+  __android_log_print(ANDROID_LOG_ERROR, "onnxruntime", "%s", msg);
+#else
+#define ORT_PRINT_ERROR_MSG(msg) \
+  std::cerr << msg << std::endl;
+#endif
+#endif
+
```
Comment (skottmckay) [inline include/onnxruntime/core/common/common.h:97]: nit: Does this need to be a macro or could it just be a helper function?

File: onnxruntime/core/common/helper.cc [hunk 2]
```diff
@@ -31,6 +43,21 @@ std::wstring ToWideString(const std::string& s) {
...
+void PrintFinalMessage(const char* msg) {
+#if defined(__ANDROID__)
+  __android_log_print(ANDROID_LOG_ERROR, "onnxruntime", "%s", msg);
+#else
+  // TODO, consider changing the output of the error message from std::cerr to logging when the
+  // exceptions are disabled, since using std::cerr might increase binary size, and std::cerr output
+  // might not be easily accesible on some systems such as mobile
+  // TODO, see if we need to change the output of the error message from std::cerr to NSLog for iOS
+  std::cerr << msg << std::endl
+            << std::flush;
```
Comment (edgchen1) [inline onnxruntime/core/common/helper.cc:58]: std::endl will flush already

File: include/onnxruntime/core/common/common.h [hunk 3]
```diff
@@ -77,6 +73,14 @@ using common::Status;
 #define ORT_ATTRIBUTE_UNUSED
 #endif
 
+#ifdef ORT_NO_EXCEPTIONS
+// Print the given final message, the message must be a null terminated char*
+// ORT will abort after printing the message.
+// For Android, will print to Android system log
+// For other platforms, will print to stderr
+void PrintFinalMessage(const char* msg);
```
[the remark this answers is not in the document]
Comment (gwang-msft) [reply include/onnxruntime/core/common/common.h:81]: Changed in my local branch, didn't commit

**Review by skottmckay — DISMISSED** [review]
:shipit:

**Review by edgchen1 — APPROVED** [review]

**Comment by fdwr** [comment]
Goodness - we cannot assume ONNX runtime consumers will be console apps (many are GUI-based) or even have a console IO stream.

**Pull request merged**